app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -0,0 +1,366 @@
+import React from "react";
+import { DataSet, Network } from "vis";
+import { forEach } from "lodash";
+import PropTypes from "prop-types";
+import ReactPanZoom from "@ajainarayanan/react-pan-zoom";
+
+import cs from "./pipeline_viz.scss";
+
+const START_NODE_ID = -1;
+const END_NODE_ID = -2;
+const STAGE_BG_COLOR = "#f8f8f8";
+const NODE_COLOR = "#eaeaea";
+const EDGE_COLOR = "#999999";
+
+class PipelineViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.stageResults = this.props.stageResults.stages;
+    this.pipelineVersion = this.props.stageResults.pipeline_version;
+    this.stageGraphsInfo = [];
+
+    if (this.props.admin) {
+      this.stageNames = [
+        "Host Filtering",
+        "GSNAPL/RAPSEARCH alignment",
+        "Post Processing",
+        "Experimental",
+      ];
+    } else {
+      this.stageNames = [
+        "Host Filtering",
+        "GSNAPL/RAPSEARCH alignment",
+        "Post Processing",
+      ];
+    }
+
+    this.state = {
+      stagesOpened: [true, true, true, true],
+      zoom: 1,
+    };
+  }
+
+  componentDidMount() {
+    this.renderGraphs();
+  }
+
+  onMouseWheelZoom(e) {
+    const zoomChange = (e.deltaY < 0 ? 1 : -1) * 0.01;
```

Comment:
for the sake of not having magic numbers, could you add a comment explaining why -0.01 and +0.01